packages/synapse-react-client/src/components/Authentication/LastLoginInfo.tsx

```diff
@@ -0,0 +1,199 @@
+  return {
+    currentSourceAppName: localStorage.getItem(
+      CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginMethod: localStorage.getItem(
+      LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    ) as LoginMethod,
+    lastLoginSourceAppName: localStorage.getItem(
+      LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginSourceAppURL: localStorage.getItem(
+      LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    ),
+    lastLoginDate: localStorage.getItem(LAST_LOGIN_DATE_LOCALSTORAGE_KEY),
+  }
+}
+
+export function setCurrentAppInfo(currentSourceAppName: string) {
+  localStorage.setItem(
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    currentSourceAppName,
+  )
+}
+
+export function setLastLoginInfo(
+  lastLoginMethod: LoginMethod,
+  lastLoginDate: string,
+  lastLoginSourceAppName: string,
+  lastLoginSourceAppURL: string,
+) {
+  localStorage.setItem(LAST_LOGIN_DATE_LOCALSTORAGE_KEY, lastLoginDate)
+  localStorage.setItem(LAST_LOGIN_METHOD_LOCALSTORAGE_KEY, lastLoginMethod)
+  localStorage.setItem(
+    LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    lastLoginSourceAppName,
+  )
+  localStorage.setItem(
+    LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    lastLoginSourceAppURL,
+  )
+}
+
+export function clearLastLoginInfo() {
+  const keys = [
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+  ]
+  keys.forEach(key => localStorage.removeItem(key))
+}
+
+function getSourceAppIndefiniteArticle(val: string) {
+  return val.match(/^[aeiou]/i) ? 'an' : 'a'
+}
+
+function getSourceAppDefiniteArticle(val: string) {
+  return val.match(' Portal$') ? 'the ' : ''
+}
+
+export default function LastLoginInfo(props: LastLoginInfoProps) {
+  const {
+    currentSourceAppName,
+    lastLoginMethod,
+    lastLoginSourceAppName,
+    lastLoginSourceAppURL,
+    lastLoginDate,
+    display,
+  } = props
+
+  const shouldNotShowLastLoginInfo =
+    currentSourceAppName === null ||
+    lastLoginMethod === null ||
+    lastLoginSourceAppName === null ||
+    lastLoginSourceAppURL === null ||
+    lastLoginDate === null ||
+    currentSourceAppName === lastLoginSourceAppName
+
+  if (shouldNotShowLastLoginInfo) {
+    return null
+  }
+
+  const currentSourceAppIndefiniteArticle =
+    getSourceAppIndefiniteArticle(currentSourceAppName)
+  const currentSourceAppDefiniteArticle =
+    getSourceAppDefiniteArticle(currentSourceAppName)
+  const lastLoginSourceAppDefiniteArticle = getSourceAppDefiniteArticle(
+    lastLoginSourceAppName,
+  )
+
+  const title = `You may already have ${currentSourceAppIndefiniteArticle} ${currentSourceAppName} account`
+
+  const tooltipText = (
+    <>
+      <Typography variant="smallText2" mb={1}>
+        {title}
+      </Typography>
+      <Typography variant="smallText1" mb={1} color="white">
```

Comment:
Set the text color to white to prevent the SageAccountWeb theme color for "smallText1" from making the tooltip text difficult to read

Reply:
I opened a ticket to track evaluating the SageAccountWeb theme overrides [here](https://sagebionetworks.jira.com/browse/PORTALS-2747)

Reply:
Thank you!